azure-mgmt-cosmosdb/src/main/java/com/microsoft/azure/management/cosmosdb/CosmosDBAccount.java

```diff
@@ -154,6 +162,61 @@
              * @return the next stage of the definition
              */
             WithConsistencyPolicy withKind(DatabaseAccountKind kind);
+
+            /**
+             * The database account kind for the CosmosDB account.
+             *
+             * @param kind the account kind
+             * @param capabilities the list of Cosmos DB capabilities for the account
+             * @return the next stage of the definition
+             */
+            @Beta(SinceVersion.V1_10_0)
+            WithConsistencyPolicy withKind(DatabaseAccountKind kind, Capability... capabilities);
+
+            /**
+             * Creates a SQL CosmosDB account.
+             *
+             * @return the next stage of the definition
+             */
+            @Method
+            @Beta(SinceVersion.V1_10_0)
+            WithConsistencyPolicy withDataModelSql();
+
+            /**
+             * Creates a MongoDB CosmosDB account.
+             *
+             * @return the next stage of the definition
+             */
+            @Method
+            @Beta(SinceVersion.V1_10_0)
+            WithConsistencyPolicy withDataModelMongoDB();
+
+            /**
+             * Creates a Cassandra CosmosDB account.
+             *
+             * @return the next stage of the definition
+             */
+            @Method
+            @Beta(SinceVersion.V1_10_0)
+            WithConsistencyPolicy withDataModelCassandra();
+
+            /**
+             * Creates an Azure Table CosmosDB account.
+             *
+             * @return the next stage of the definition
+             */
+            @Method
+            @Beta(SinceVersion.V1_10_0)
+            WithConsistencyPolicy withDataModelAzureTable();
```

Comment:
>withDataModelAzureTable [](start = 34, length = 23)

Should it be just withDataModel method and then a parameter of Enum type with all the supported DB Types and enum members? Too many methods in Intelisence that start with the same prefix (A long one in this case) creates a bunch of clutter IMHO

Reply:
also it is kind of weird construct `withDataModelSql` . More natural would be `withSqlAccount` or `withSqlTyoe` but in that case you will have types in a hard to find order in the intelisence. So reiterating my proposal to have one method and bunch of enum values :)

Reply:
In the management portal it is called API 
![cosmosdbportalui](https://user-images.githubusercontent.com/3217895/39220566-c6a6e1ca-47e6-11e8-9f31-ac6750c550e2.png)

Reply:
We do not need to match exactly what the Azure Portal does. Data model fits better the type of the account (see the explanation behind the little "i").

There's already a method that takes enums (see the withKind() overload I just added). These new methods are direct shortcuts for creating the respective CosmosDB account.